packages/pg/lib/result.js

```diff
@@ -59,8 +59,7 @@ class Result {
     return row
   }
 
-  parseRow(rowData) {
```

Comment:
I don’t know if this API change is considered important, but how about prebuilding in `Result`’s constructor? I think it’s cleaner anyway (`Result` is already responsible for `rowMode`, for example), and then `parseRow`’s signature would be able to stay exactly the same. Also, no lazy-building branch if it prebuilds unconditionally in the constructor, no potentially unwanted shared state between the results of multi-result queries, and we could even drop `} else { row[field] = null }` below.

Reply:
Regarding moving logic to Result; Result is initialized from the constructor of Query; I think field data is only available after handleRowDescription is called; so then I would modify/append to Result after it being constructed - in my eyes I would potentially introduce some possibilities of failure of edge-cases by doing this in another place. If you are sure that it won't  I can try, but I would first have to get my test-setup running steadily. 

I had the null omission in my original patch which I didn't pull, if we are not making it optional then that makes sense indeed.

Reply:
Oh; I can just make Result responsible for having a template row instead of query; that makes sense, working on it!

Reply:
Pushed a commit which moves all the logic to Result; which is indeed even more clean! Also removed the null part

Reply:
Nice. Right, sorry, no row description yet in the constructor – but how about in `addFields`? In the same loop as the parser assignment, even.

Reply:
I read over addFields as being potentially useful, as normally I would name what it does as 'setFields'; but indeed, this seems like the best place to avoid an extra check in the often-called-function. Committed